### ISSUE TITLE
Embedded videos are missing after upgrade to Liveblog 1.5

### ISSUE THREAD

**Issue by budzanowski** [Automattic/liveblog#235]
When user adds video e.g. from Youtube it is missing from the post.

Steps to reproduce:
1. Add link to youtube video.
2. Publish
3. lembedded video frame is missing.

Update 1.5 improved escaping by adding sanitation functions at output.
Post rendered by wp_embed is if filtered through wp_kses_post that cuts iframe tag.

@philipjohn  would it be a good idea to add iframe to allowedposttags?

/cc @nb, added for trial visibility.

**Comment by philipjohn** [comment]
That sounds like a good approach, @budzanowski . It looks like we can't add to to `wp_kses_post()`'s allowed tags, though.

**Comment by budzanowski** [comment]
We could pass it along with other allowed tags to wp_kses. If we additionally change the order of how things work and apply wp_kses after rendering the content in render_content function we could output directly. This way we only do the wp_kses once per added liveblog post.

**Comment by budzanowski** [comment]
After carefully examining the issue I have changed my approach. I have created PR with this new approach - #237.

**Comment by philipjohn** [comment]
@budzanowski While I understand the reasoning here, moving the `wp_kses_post()` renders it mostly redundant - the purpose of that is to escape the output, and we should do that as late as possible: https://vip.wordpress.com/2014/06/20/the-importance-of-escaping-all-the-things/

Given we can reasonably predict what the HTML will look like we can easily pass in a list of allowed tags and attributes to `wp_kses()`.

**Comment by budzanowski** [comment]
@philipjohn I understand why we want to do `wp_kses_post` as late as possible. Part of my initial idea (second comment) was exactly what you have suggested - to pass list of tags. I have abandoned this idea when i have realized that it introduces additional dependency between how we use `oembed` and `wp_kses`. We would have to maintain manually a list of allowed tags(not only tags, also attributes) by hand,or have a script that we would run from time to time that would. Without this whenever there is a change in `shortcodes` or `oembed` behavior for particular site or additional site gets whitelisted in WP for `wp_embed` we risk failure in displaying the content properly. 

Additionally we will allow user to input a lot of potentially dangerous tags because we would allow them in `wp_kses` with our allowed tags. We can address this by using  `wp_kses_post` in render the way it is now and add `wp_kses'  with allowed tags at the end. The only problem here it is that it does not solve the issue I have explained in previous paragraph and adds additional performance requirements.

I have no problem going your way. I am just concerned that this may not be sufficient for proper work of the plugin. Could you pleas address my concerns? :)

**Comment by philipjohn** [comment]
Thanks for thinking this through @budzanowski ! Those are really valid points.

> We would have to maintain manually a list of allowed tags(not only tags, also attributes) by hand

I think this is okay - we can be reasonably confident what the HTML is going to look like, and therefore what tags and attributes are required.

> Additionally we will allow user to input a lot of potentially dangerous tags because we would allow them in wp_kses with our allowed tags.

This is true. At the same time, we know we're outputting that HTML anyway, and we're not taking away from escaping efforts elsewhere, so this is a risk we've already accepted.

> The only problem here it is that it does not solve the issue I have explained in previous paragraph and adds additional performance requirements.

We should be fine with a single call to `wp_kses()` towards the end, as that fulfils our escaping need.

> I have no problem going your way. I am just concerned that this may not be sufficient for proper work of the plugin.

We should perhaps have some tests :) Is there something particular you're thinking of, or being cautious?

**Comment by budzanowski** [comment]
@philipjohn thanks for clarifications.
It looks like the additional `wp_kses_post` I was proposing is in fact redundant. To be absolutely sure I have checked how the user input is handled end-to-end. It turns out that 'wp_kses' in form of `wp_filter_post_kses` is already used in `insert` and `update` methods from `WPCOM_Liveblog_Entry` class. This means that `render` already gets sanitized input.

To be sure that it will stay this way I have added a test that checks that Liveblog Entry holds sanitized content. I check for following tags: iframe, script, applet, embed, object.
